keras_cv/layers/preprocessing/random_rotation_test.py

```diff
@@ -0,0 +1,122 @@
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomRotation(0.5)
+        actual_output = layer(input_images, training=False)
+        self.assertAllClose(expected_output, actual_output)
+
+    def test_distribution_strategy(self):
+        """Tests that RandomRotation can be created within distribution
+        strategies."""
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        strat = tf.distribute.MirroredStrategy(devices=["cpu", "gpu"])
+        with strat.scope():
+            layer = RandomRotation(0.5)
+            output = strat.run(lambda: layer(input_images, training=True))
+        values = output.values
+        self.assertAllEqual(2, len(values))
+
+    def test_config_with_custom_name(self):
+        layer = RandomRotation(0.5, name="image_preproc")
+        config = layer.get_config()
+        layer_1 = RandomRotation.from_config(config)
+        self.assertEqual(layer_1.name, layer.name)
+
+    def test_unbatched_image(self):
+        input_image = np.reshape(np.arange(0, 25), (5, 5, 1)).astype(np.float32)
+        # 180 rotation.
+        layer = RandomRotation(factor=(0.5, 0.5))
+        output_image = layer(input_image)
+        expected_output = np.asarray(
+            [
+                [24, 23, 22, 21, 20],
+                [19, 18, 17, 16, 15],
+                [14, 13, 12, 11, 10],
+                [9, 8, 7, 6, 5],
+                [4, 3, 2, 1, 0],
+            ]
+        ).astype(np.float32)
+        expected_output = np.reshape(expected_output, (5, 5, 1))
+        self.assertAllClose(expected_output, output_image)
+
+    def test_augment_image(self):
+        input_image = np.reshape(np.arange(0, 25), (5, 5, 1)).astype(np.float32)
+        # 180 rotation.
+        layer = RandomRotation(factor=(0.5, 0.5))
+        output_image = layer.augment_image(
+            input_image, transformation=layer.get_random_transformation()
+        )
+        expected_output = np.asarray(
+            [
+                [24, 23, 22, 21, 20],
+                [19, 18, 17, 16, 15],
+                [14, 13, 12, 11, 10],
+                [9, 8, 7, 6, 5],
+                [4, 3, 2, 1, 0],
+            ]
+        ).astype(np.float32)
+        expected_output = np.reshape(expected_output, (5, 5, 1))
+        self.assertAllClose(expected_output, output_image)
+
+    def test_augment_bbox(self):
+        input_image = np.random.random((512, 512, 3)).astype(np.float32)
+        bboxes = tf.convert_to_tensor([[200, 200, 400, 400], [100, 100, 300, 300]])
+        # 180 rotation.
+        layer = RandomRotation(factor=(0.5, 0.5), bounding_box_format="xyxy")
+        output_bbox = layer.augment_bounding_boxes(
+            input_image,
+            bboxes,
+            transformation=layer.get_random_transformation(),
+        )
+        expected_output = np.asarray(
+            [[111, 112, 312, 312], [212, 211, 412, 412]]
```

Comment:
I was somehow expecting the x/y are the same after rotation, is it because of int casting?

Reply:
there is layer and layer_1

Reply:
I am not sure I understand, can you explain a little bit more?

Reply:
I mean the result after rotation should be [111, 111, 312, 312]? Not sure why there is a off by one here for to corner.

Reply:
Oh yes, this could be because of  int casting